.github/workflows/push-docker.yaml

```diff
@@ -61,3 +69,57 @@ jobs:
         if: ${{ ! inputs.dry-run }}
         run: |
           docker push --all-tags "${{ inputs.repository-name }}"
+  push-docker-multi-arch:
+    name: Push Semgrep Docker Image (multi-arch)
+    runs-on: ubuntu-22.04
+    if: ${{ inputs.multi-arch }}
+    steps:
+      - uses: iarekylew00t/regctl-installer@v1
+      - uses: actions/download-artifact@v3
+        with:
+          path: /tmp/artifacts
+      - run: |
+          artifact_filenames=(/tmp/artifacts/${{ inputs.artifact-name }}-arch-*/image.tar)
+          if [[ ${#artifact_filenames[@]} == 0 ]]; then
+            echo "No artifacts found matching ${{ inputs.artifact-name }}-arch-*"
+            exit 1
+          fi
+          mkdir -p /tmp/merged/blobs/sha256
```

Comment:
A couple comments to explain the necessity and behavior of these more complex steps would probably be helpful for someone who has to touch this in the future so they don't have to dig up this PR.

Reply:
Added some comments and refactored things a bit so that we can flip back to non-depot single-arch builds if necessary